File: abc146/src/bin/c.rs
use proconio::{fastout, input};

#[fastout]
fn main() {
    input! {
        a: u64, b: u64, x: u64,
    }
    let (mut lo, mut hi) = (0, x.min(1_000_000_001));
    while lo < hi {
        let m = (lo + hi) / 2;
        if a * m + b * m.to_string().len() as u64 > x {
            hi = m;
        } else {
            lo = m + 1;
        }
    }
    println!("{}", lo.max(1) - 1);
}

File: abc146/c.ml
open Base;;

let a, b, x =
  Caml.Scanf.sscanf (Caml.read_line ()) "%d %d %d" (fun a b x -> (a, b, x))
in
let answer =
  List.init 10 ~f:(fun i ->
      (x - (b * (i + 1))) / a |> min ((10 ** (i + 1)) - 1))
  |> List.fold ~init:0 ~f:max
  |> min (10 ** 9)
in
answer |> Int.to_string |> Caml.print_endline
